### ISSUE TITLE
Clone only latest revision from git repository

### ISSUE THREAD

**Issue by madoar** [PhoenicisOrg/phoenicis#1720]
Currently when cloning a git repository we clone the whole repository, which are in total around 59mb at the moment. This process takes some time.

To increase the performance and save some space I propose to only clone and later on fetch the latest revision of the repository and not the whole history.

**Comment by plata** [comment]
What if you change the branch in the repository settings?

**Comment by madoar** [comment]
You mean in Phoenicis?
It gets a new hashcode and therefore a new folder in the `cache` directory.

**Comment by plata** [comment]
And it's also possible to pull in a shallow clone? Just asking, I never tried that.

**Comment by qparis** [comment]
Do we still have a HTTP interface? 
Maybe we could create a small server-side app that caches the git branch you need

**Comment by madoar** [comment]
There seems to be a way to only clone the latest revision via `git clone --depth=1 <remote_repo_url>` (see https://stackoverflow.com/questions/1209999/using-git-to-get-just-the-latest-revision), but I haven't tried it yet.

**Comment by plata** [comment]
Yes, cloning like this works. I'm just not sure if you can update that clone later.

**Comment by madoar** [comment]
Additionally there seem to be system variables, wine configuration variables and other things.
I'm not sure how much effort it is to translate/convert these things.

**Comment by madoar** [comment]
How is the performance and the memory usage?
Is it faster compared to cloning the whole repository?
Does it take a lot less memory?

**Comment by plata** [comment]
I would say that this is slower if the repository is changed frequently (whatever "frequently" means here).

**Comment by qparis** [comment]
Except if you also cache the miniatures

**Comment by plata** [comment]
I don't get that part.

**Comment by qparis** [comment]
In POLv4, only the txt data is redownload each time. The miniatures are cachée locally even after a reload

**Comment by qparis** [comment]
The request are made separately

**Comment by plata** [comment]
Understand. But that should still be more compared to a git pull.

**Comment by madoar** [comment]
I think what @qparis is suggesting is the addition of a new `WebRepository`, which I'm in favor of.
The backend implementation could be a frontend for git, which I think is a great idea!!!

I also believe that solving this issue can lead to a huge increase in the performance of Phoenicis, because the loading time for all the stuff contained inside the repositories is quite huge in my opinion.

Additionally we could think about splitting the engines and the applications into separate repositories.
This could also improve the loading time and decouple the engines tab from the applications tab.
